drivers/console/uart_mcumgr.c

```diff
@@ -185,6 +242,10 @@ void uart_mcumgr_register(uart_mcumgr_recv_fn *cb)
 	uart_mcumgr_dev = device_get_binding(CONFIG_UART_MCUMGR_ON_DEV_NAME);
 
 	if (uart_mcumgr_dev != NULL) {
+#if defined(CONFIG_MCUMGR_SMP_UART_ASYNC)
```

Comment:
if you would use `if (IS_ENABLED(CONFIG_MCUMGR_SMP_UART_ASYNC))` here you could get rid of all ifdefs above.

Reply:
I have removed the #ifdef and use the same name for setup functions; different function gets compiled when async is enabled.